src/index/updater/inscription_updater/stream.rs

```diff
@@ -389,3 +400,50 @@ impl StreamEvent {
     Ok(())
   }
 }
+
+pub struct CacheEntry<V> {
+  value: V,
+  expires_at: Instant,
+}
+
+pub struct Cache<K, V> {
+  map: Mutex<HashMap<K, CacheEntry<V>>>,
+  ttl: Duration,
+}
+
+impl<K, V> Cache<K, V>
+where
+  K: Eq + std::hash::Hash + Clone,
+  V: Clone,
+{
+  fn new(ttl: Option<Duration>) -> Cache<K, V> {
+    let ttl = ttl.unwrap_or(Duration::from_secs(60)); // Default to 1 minute
+    Cache {
+      map: Mutex::new(HashMap::new()),
+      ttl,
+    }
+  }
+
+  pub fn set(&self, key: K, value: V) {
+    let mut map = self.map.lock().unwrap();
+    let entry = CacheEntry {
+      value,
+      expires_at: Instant::now() + self.ttl,
+    };
+    map.insert(key, entry);
+  }
+
+  pub fn get(&self, key: &K) -> Option<V> {
+    let mut map = self.map.lock().unwrap();
+    let now = Instant::now();
+    map.retain(|_, v| v.expires_at > now); // Clean up expired entries
```

Comment:
this operation can be slow, can we only do this on `set` only and/or when cache grows above certain size?